contracts/L1Teleporter.sol

```diff
@@ -140,11 +144,10 @@ contract L1Teleporter is Pausable, AccessControl, L2ForwarderPredictor, IL1Telep
         if (teleportationType == TeleportationType.Standard) {
             // standard type requires 1 retryable to L3 paid for in ETH
             ethAmount += costs.l2l3TokenBridgeCost;
-            feeTokenAmount = 0;
         } else if (teleportationType == TeleportationType.OnlyCustomFee) {
             // only custom fee type requires 1 retryable to L3 paid for in fee token
             feeTokenAmount = costs.l2l3TokenBridgeCost;
-        } else {
+        } else if (params.l3FeeTokenL1Addr != SKIP_FEE_TOKEN_MAGIC_ADDRESS && costs.l2l3TokenBridgeCost != 0) {
```

Comment:
```suggestion
        } else if (costs.l2l3TokenBridgeCost > 0) {
```
and then we add a check for at the beginning of the entrypoint where if `l3FeeTokenL1Addr` is the magic address, all l3 fee related params should be 0

Reply:
[3ea7eab](https://github.com/OffchainLabs/l1-l3-teleport-contracts/pull/30/commits/3ea7eabe98a21172553669658a881d21ba68f7fb)